backend/substrapp/tasks/tasks.py

```diff
@@ -625,6 +627,41 @@ def _do_task(client, subtuple_directory, tuple_type, subtuple, compute_plan_id,
             client.volumes.create(name=volume_id)
         model_volume[volume_id] = {'bind': '/sandbox/local', 'mode': 'rw'}
 
+    if compute_plan_id is not None and settings.TASK['CHAINKEYS_ENABLED']:
+        chainkeys_directory = get_chainkeys_directory(compute_plan_id)
+
+        if not os.path.exists(chainkeys_directory):
+            secret_namespace = os.getenv('K8S_SECRET_NAMESPACE', 'default')
+            label_selector = f"compute_plan={subtuple.get('tag')}"
+
+            k8s_client = get_k8s_client()
+            try:
+                secrets = k8s_client.list_namespaced_secret(secret_namespace, label_selector=label_selector)
+            except ApiException as e:
+                logging.error(f'failed to fetch namespaced secrets {secret_namespace} with selector {label_selector}')
+                raise e
+
+            secrets = [{s['metadata']['name']: int.from_bytes(b64decode(s['data']['key']), 'big')}
+                       for s in secrets.to_dict()['items']]
```

Comment:
After some thinking,

```python
secrets = {s['metadata']['name']: int.from_bytes(b64decode(s['data']['key']), 'big') for s in secrets.to_dict()['items']}
```
seems better as it will create a dict instead of a list of dict.
Will be easier then for accessing secrets.
Example:
```python
secrets['name_of_secret']
```

Reply:
updated 531f9b2